app/scripts/controllers/locationPickerMap.ctrl.js

```diff
@@ -80,109 +88,36 @@ angular.module('yourStlCourts').controller('LocationPickerMapCtrl', function ($s
-    }
-
+  function getMunicipalityFromMapName(muniMapName){
     function getName(nameString){
-      nameString = capitalizeFirstLetter(nameString);
-      if (nameString == "Unincorporated"){
-        nameString = "St. Louis County";
+      if (nameString.toLowerCase() == "unincorporated"){
+        nameString += " St. Louis County";
       }
-      return nameString;
-    }
 
-    return {
-      id: muniId,
-      name: getName(muniName),
-      isEqualTo: function (muniName) {
-        return (this.name == muniName.toLowerCase());
-      },
-      getDatabaseMuniEntry: function(){
-        //lookup the entry from Municipalities
-        var dbEntry = null;
-        var friendlyMuniName = this.name.replace("&","and");
-        for (var i in municipalities){
-          var dbMuniName = municipalities[i].municipality_name;
-          if (friendlyMuniName == "St. Louis County"){
-            dbEntry = {municipality_name:friendlyMuniName};
-            break;
-          }
-          if (dbMuniName.toLowerCase() == friendlyMuniName.toLowerCase()){
-            dbEntry = municipalities[i];
-            break;
-          }
-        }
-        return dbEntry;
-      }
+      return nameString.replace("&", "and");
     }
+
+    return _.find(municipalities, function(municipality) {
+      return municipality.name.toLowerCase() === getName(muniMapName).toLowerCase();
+    });
   }
 
-  ctrl.selectedMunicipalitiesObj = {
-    municipalityObjs: new Array(),
-    municipalityClicked: function(municipalityObj){
-      var municipalityAdded = false;
-      if (this.idInArray(municipalityObj.id)){
-        this.removeMunicipalityObj(municipalityObj.id);
-      }else{
-        this.addMunicipalityObj(municipalityObj);
-        municipalityAdded = true;
-      }
-      return municipalityAdded;
-    },
-    addMunicipalityObj: function(municipalityObj){
-      this.municipalityObjs.push(municipalityObj);
-    },
-    removeMunicipalityObj: function (municipalityId) {
-      for(var index in this.municipalityObjs){
-        if (this.municipalityObjs[index].id == municipalityId){
-          this.municipalityObjs.splice(index,1);
-        }
-      }
-    },
-    getArrayOfNames: function(){
-      //this function removes duplicate names while allowing the selectedMunicipalitesObj to retain duplicated names with different ids
-      var arrayOfNames = new Array();
-      var countyInArray = false;
-      for(var index in this.municipalityObjs){
-        arrayOfNames.push(this.municipalityObjs[index].name);
-        if (this.municipalityObjs[index].name == "St. Louis County"){
-          if (!countyInArray){
-            countyInArray = true;
-          }else{
-            arrayOfNames.pop();  //only need one entry for county so don't add again.
-          }
+  function isMunicipalitySelected(municipalityId) {
+    return !!_.find(ctrl.selectedMunicipalities, {id: municipalityId});
```

Comment:
It's more resilient than checking explicitly for undefined because if you change the lookup to something that doesn't return undefined, then your function no longer works. If it's basing things on truthy and falsy, then you're safer from future changes.